pkg/installation/semver/version_test.go

```diff
@@ -0,0 +1,107 @@
+// Copyright 2019 The Kubernetes Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package semver
+
+import (
+	"fmt"
+	"testing"
+)
+
+func TestParse(t *testing.T) {
+	cases := []struct {
+		name    string
+		in      string
+		want    string
+		wantErr bool
+	}{
+		{"empty", "", "", true},
+		{"zero is valid", "v0.0.0", "v0.0.0", false},
+		{"no v prefix", "v1.0.0", "v1.0.0", false},
+		{"spaces between v and value", "v 1.0.0", "", true},
+		{"major", "v1", "", true},
+		{"major.minor", "v1.2", "", true},
+		{"major.minor.patch", "v1.2.3", "v1.2.3", false},
+		{"major.minor.patch-suffix", "v1.2.3-beta.2+foo.bar", "v1.2.3-beta.2+foo.bar", false},
+		{"leading zero in major", "v01.2.3", "", true},
+		{"leading zero in minor", "v1.02.3", "", true},
+		{"leading zero in patch", "v1.2.03", "", true},
+		{"negative value in major", "v-1.2.3", "", true},
+		{"negative value in minor", "v1.-2.3", "", true},
+		{"negative value in patch", "v1.2.-3", "", true},
+		{"empty pre-release identifier", "v1.0.1-", "", true},
+		{"empty meta identifier", "v1.0.1+", "", true},
```

Comment:
Maybe also worth testing if `[:alpha:]` chars lead to errors.

Reply:
can you give a specific example?

Reply:
Something like
```go
{"major with alpha chars", "va.0.1", "", true},
{"minor with alpha chars", "v1.a.1", "", true},
{"patch with alpha chars", "v1.0.a", "", true},
```

Reply:
that feels a bit too obvious but I'll add tests for `v0a.0.0`, `v0.0a.0` and `v0.0.0a`